test/tests/misc/check_error/tests

```diff
@@ -343,7 +343,7 @@
   [./missing_req_par_moose_obj_test]
     type = 'RunException'
     input = 'missing_req_par_moose_obj_test.i'
-    expect_err = "missing required parameter 'Kernels/diff/variable'"
+    expect_err = "Cannot add Kernel for variable , it is not a nonlinear variable"
```

Comment:
This does not seem right. The test tests that if people forget a variable parameter, we tell them to add it.  The new error message does not really implies that.   When is the old error message triggered?

Reply:
Hmm... "variable" is a `RequiredParam` on `KernelBase`, so if the user forgets to provide it, the original error message is triggered in `InputParameters::checkParams()`, which must be happening after the new sanity check. I suppose I could change the sanity check to something involving `parameters.isParamValid("variable")` as well?